platform/include/reconnect.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file reconnect_config.h
+ * @brief Declaration of the exponential backoff reconnect logic utility functions
+ * and constants.
+ */
+
+
+/* bools are only defined in C99+ */
+#if defined( __cplusplus ) || __STDC_VERSION__ >= 199901L
+    #include <stdbool.h>
+#elif !defined( bool )
+    #define bool                             signed char
+    #define false                            0
+    #define true                             1
+#endif
+
+/* @brief Max number of connect attempts, set this value to 0 if the client
+ * must try connecting to the server forever */
+#define MAX_RECONNECT_ATTEMPS                4U
+/* @brief Initial fixed timeout value in seconds between two successive
+ connects. A random jitter value is added to every timeout value  */
+#define RECONNECT_INITIAL_TIMEOUT_SECONDS    1U
+/* @brief Max timout value in seconds */
+#define MAX_RECONNECT_TIMEOUT                32U
+/* @brief Max jitter value in seconds */
+#define MAX_JITTER_VALUE_SECONDS             5U
```

Comment:
If a customer has a large fleet of devices, we will expect them to modify these values, right? How would a customer know what value to choose for their fleet? 

And how would a customer discover that these values need to be updated for their fleet.

Reply:
Good point, we will have these values from the existing SDKs as default.  ( I'll update those ) This example code is making sure that the customers do use some kind of exponential backoff mechanism along with jitter for re-connection, but if required they will have to fine tune it based on their use case.

Reply:
I have updated the max value ( from 32 to 128 seconds) as per V3 CSDK. 
https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/d039f075e1cc2a2a7fc20edc6868f328d8d36b2f/samples/linux/subscribe_publish_library_sample/aws_iot_config.h#L53